src/internet_identity/src/storage/anchor.rs

```diff
@@ -164,6 +167,20 @@ impl Anchor {
         &self.devices
     }
 
+    /// Returns matching [Credential]s.
```

Comment:
nit: maybe add a description of what happens when `filter` is `None`?
Alternatively, maybe drop `Option`, and leave it up to the caller to provide the dummy "catch-all" filter?  Maybe then also `Box` would be not needed?

Reply:
Yeah, I think the whole interface gets a lot nicer if the `filter` is mandatory.